modules/otel-collector-components.adoc

```diff
@@ -870,6 +870,46 @@ This extension supports traces, metrics, and logs.
 <11> Sets a timeout for the token client's request.
 <12> You can assign the authenticator configuration to an OTLP exporter.
 
+[id="filestorage-extension_{context}"]
+=== File Storage Extension
+
+:FeatureName: The File Storage Extension
+include::snippets/technology-preview.adoc[]
+
+The File Storage Extension supports traces, metrics, and logs. This extension can persist the state to the local file system. This extension persists the sending queue for the OTLP exporters that are based on the HTTP and the gRPC protocols. This extension requires the read and write access to a directory. This extension can use a default directory, but the default directory must already exist.
+
+.OpenTelemetry Collector custom resource with a configured File Storage Extension that persists an OTLP sending queue
+[source,yaml]
+----
+  config: |
+    extensions:
+      file_storage/all_settings:
+        directory: /var/lib/otelcol/mydir # <1>
+        timeout: 1s # <2>
+        compaction:
+          on_start: true # <3>
+          directory: /tmp/ # <4>
+          max_transaction_size: 65_536 # <5>
+        fsync: false # <6>
+
+    exporters:
+      otlp:
+        sending_queue:
+          storage: file_storage/all_settings # <7>
+
+    service:
+      extensions: [file_storage/all_settings]
+      pipelines:
+        traces:
+          receivers: [otlp]
+          exporters: [otlp]
+----
+<1> Specifies the directory in which the telemetry data is stored.
+<2> Specifies the timeout time interval for opening the stored files.
+<3> Starts compaction when the Collector starts. If omitted, the default is `+false+`.
+<4> Specifies the directory in which the compactor stores the telemetry data.
+<5> Defines the maximum size of the compaction transaction. To ignore the transaction size, set to zero. If omitted, the default is `+65536+`.
+<6> When set, forces the database to perform an `fsync` call after each write operation. This helps to ensure database integrity if there is an interruption to the database process, but at the cost of performance.
```

Comment:
You are missing a callout for `<7>`.

![Screenshot 2024-05-23 at 2 51 19 PM](https://github.com/openshift/openshift-docs/assets/96192119/7ac73189-64f3-40ca-ac0a-f69951ec976f)